addon_imps/storage/azure_blob_storage.py

```diff
@@ -0,0 +1,211 @@
+                    )
+                )
+        return containers
+
+    def _parse_blob_items(
+        self,
+        xml_root: ET.Element,
+        container_name: str,
+        prefix: str,
+        item_type: Optional[storage.ItemType],
+    ) -> list[storage.ItemResult]:
+        items = []
+
+        for blob_prefix in xml_root.findall(".//BlobPrefix"):
+            name_elem = blob_prefix.find("Name")
+            if name_elem is not None and name_elem.text:
+                folder_path = name_elem.text
+                folder_name = folder_path.rstrip("/").split("/")[-1]
+                item_result = storage.ItemResult(
+                    item_id=f"{container_name}:/{folder_path}",
+                    item_name=f"{folder_name}/",
+                    item_type=storage.ItemType.FOLDER,
+                )
+                if item_type is None or item_result.item_type == item_type:
+                    items.append(item_result)
+
+        for blob in xml_root.findall(".//Blob"):
+            name_elem = blob.find("Name")
+            if name_elem is not None and name_elem.text:
+                blob_name = name_elem.text
+
+                if prefix:
+                    if not blob_name.startswith(prefix):
+                        continue
+                    relative_name = blob_name[len(prefix):]
+                else:
+                    relative_name = blob_name
+
+                if "/" not in relative_name:
+                    item_result = storage.ItemResult(
+                        item_id=f"{container_name}:/{blob_name}",
+                        item_name=relative_name,
+                        item_type=storage.ItemType.FILE,
+                    )
+                    if item_type is None or item_result.item_type == item_type:
+                        items.append(item_result)
+        return items
+
+    async def list_root_items(self, page_cursor: str = "") -> storage.ItemSampleResult:
+        try:
+            async with self.network.GET(
+                "?comp=list",
+                headers=self._get_api_headers(),
+                query={"maxresults": self.MAX_RESULTS},
+            ) as response:
+                xml_root = ET.fromstring(await response.text_content())
+                return storage.ItemSampleResult(items=self._parse_containers(xml_root))
+        except Exception as e:
+            logger.error(f"Failed to list containers: {str(e)}")
+            raise UnexpectedAddonError("Failed to list containers")
+
+    async def list_child_items(
+        self,
+        item_id: str,
+        page_cursor: str = "",
+        item_type: Optional[storage.ItemType] = None,
+    ) -> storage.ItemSampleResult:
+        if not self._is_valid_item_id(item_id):
+            logger.warning(f"Invalid item ID format: {item_id}")
+            return storage.ItemSampleResult(items=[])
+
+        if ":" not in item_id:
+            return storage.ItemSampleResult(items=[])
```

Comment:
Should these cases raise some error instead of returning an empty array?

Reply:
Refactored item_id parsing and added error handling for invalid values